### DISCUSSION TITLE
API: Get the globally unique ID of an account

### DISCUSSION THREAD

**Discussion by bitbonk** [mastodon/mastodon#22732] [Q&A] [answered]
For a bot that uses the mastodon HTTP APIs and that is supposed to boos certain posts, I want to make sure that a post was not actually created by the bot itself.

To get the account information of the bot account I am using [verify_credentials](https://docs.joinmastodon.org/methods/accounts/#verify_credentials).

My question now is, is it safe to use the `"id"` json property I get from that method and reliably compare it with `"account": { "id" ... }` of a status. In other words is that `"id"` property guaranteed to be always unique globally, across all instances? If not, what json property should I use instead?

**Comment by rbairwell** [comment]
No, that id is local to that instance. You should use the URN/full handle (username) to identify accounts across the fediverse.

**Comment by trwnh** [comment, marked as answer]
`id` is guaranteed to be *locally* unique, which isn't a problem because your bot should only be connecting to the local Mastodon API.

**Reply by bitbonk** [reply]
But the bot is going to inspect posts from the federated timeline not just the local one so I'll something globally unique.

**Reply by trwnh** [reply]
@bitbonk Nope, it's fine -- remote posts and accounts still have a locally unique `id`

**Reply by bitbonk** [reply]
So on an instance, all users even if they come from other instances will have a unique ID? 
And the same user may have another ID on another instance?

**Reply by trwnh** [reply]
@bitbonk Yeah, the API returns a local database ID for everyone, no matter where they came from. For example, `gargron@mastodon.social` is account `1` on mastodon.social but he's account `2` on mastodon.online. Account `1` on mastodon.online is `gargron@mastodon.online`.

The important bit is that if you're using the API, all ids are unique to the instance whose API you are using.
